### PR TITLE
Fix Muscle Memory

### PR DESCRIPTION
**Pull request by hannibal002**
Reverted Hypiyel's latest change that moved many buttons from the SkyBlock Menu into ***** sub categories.

grafik

### CODE REVIEW

**Comment by hannibal002** [comment]
FML


grafik

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+                is SkullItemData -> Utils.createSkull(name, itemData.uuid, itemData.value, array)
+
+                else -> {
+                    throw Error("item data error!")
+                }
+            }
+            cachedItem = result
+            return result
+        }
+    }
```
Comment (romangraef) [inline src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt:173]: Maybe: 
```kt
val item by lazy {}
```

Comment (hannibal002) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt:173]: need a ted talk for how tpo use lazy

Comment (romangraef) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt:173]: It's essentially what you did:

```kt
val someVariable by lazy {/*some expensive function*/} 
// Does not calculate function yet.

// Later...

println(someVariable) // Now expensive function gets called.

// Even Later...

println(someVariable) // Cached Value used

```

its a really simple thing.

in theory you can also specify a thread safety mode (with the default being thread safe = true), but nobody does that. (see `lazy(LazyThreadSafetyMode.PUBLICATION) {}` or similar if you care)

Comment (romangraef) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt:173]: It's a bit like `Supplier.memoized` from guava, but with kotlin `val by` syntax easier available and thread safety (optionally) enabled.

Comment (romangraef) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt:173]: Oh while we are on the topic of caching: maybe have a lookup table from slotId -> replacment item so you dont have to keep iterating so much.

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt [hunk 2]
```diff
@@ -147,29 +146,25 @@ object OldSkyBlockMenu {
...
             }
             lore.add("")
             lore.add("§eClick to execute /$command")
             val array = lore.toTypedArray()
-
             val name = "§a$displayName"
-            val result = when (itemData) {
+            when (itemData) {
                 is NormalItemData -> Utils.createItemStackArray(itemData.displayIcon, name, array)
                 is SkullItemData -> Utils.createSkull(name, itemData.uuid, itemData.value, array)
```
Comment (romangraef) [inline src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt:160]: you can make `ItemData` a `sealed interface` which allows you to drop the extra else branch

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt (hunk 1 above)
Comment (hannibal002) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt]: implemented lazy stuff and map stuf and janis hot-loved booster cookie stuff. now im done ig.

**Comment by NopoTheGamer** [comment]
image
feature doesnt work if you dont have booster cookie warning on, maybe make clicking on it run the command even if you don't have a cookie just in case

**Comment by NopoTheGamer** [comment]
Personally I think cookie should go where close is, put sacks in the right spot and add /effects. Also if you wanted to go wild you could add back /savethejerrys, and the 2 buttons bottom left if they have commands

**Comment by hannibal002** [comment]
Jani wanted no /effects button and for /savethejerrys I would need to calculate if its jerry season or not (not want to check the sidebar for "winter" word, but real skyblock time info with "closes in x" and "opens in x" info added dynamically too

File: src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt (hunk 2 above)
Comment (hannibal002) [reply src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/OldSkyBlockMenu.kt]: Sealed is really funny

**Comment by hannibal002** [comment]
Have not touched jerry island or potion effects.

**Review by romangraef — DISMISSED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Comment by hannibal002** [comment]
I found an error when switching profile. idk if it's important enough to stop merging this PR tho. Outside of bingo time it should not happen too often and the usefulness of the pr is imo higher than the small bug.

**Comment by NopoTheGamer** [comment]
> I found an error when switching profile. idk if it's important enough to stop merging this PR tho. Outside of bingo time it should not happen too often and the usefulness of the pr is imo higher than the small bug.

what does "an error" entail

**Comment by hannibal002** [comment]
> what does "an error" entail

Not correctly detecting if the booster cookie effect is active or not.

**Review by romangraef — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Comment by hannibal002** [comment]
Lulo when you approve & merge this PR, please open #588 as ready for review :)

**Review by Lulonaut — APPROVED** [review]

**Pull request merged**